core/templates/services/contextual/url.service.ts

```diff
@@ -120,9 +120,8 @@ export class UrlService {
           constants.VALID_URL_FRAGMENT_REGEX)) {
         return this.getUrlParams().topic_url_fragment;
       }
-    } else {
-      throw new Error('Invalid URL for topic');
     }
+    throw new Error('Invalid URL for topic');
```

Comment:
@EeshaArif Removing this else block does change the behavior of this function.

Previously, if the statement on line 117 evaluates to false, the function ends there.

But now the error is thrown.